### PR TITLE
Feat sticky nav for new plan a lesson - LESQ-782

### PR DESCRIPTION
**Pull request by JBR90**
## Description

Music year: {owa_music_year}

- Adds plan a lesson page nav to page
- Connects to cms
- Crudely adds some content without styling

## Acceptance Criteria:

- [x]  Using “sanity data” create sticky contents menu
- [x]  Ensure bullet-point style exists in Oak Components
- [x]  On mobile it is just displayed at the top
- [x]  Click an item in the contents menu and it takes you to the section
- [x]  Focus moves to the section you have selected
- [x]  At the bottom of each section there is a “Return to contents” link

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2416--oak-web-application.netlify.thenational.academy/lesson-planning-new
- Test nav and back to contents buttons

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/__tests__/pages/lesson-planning-new.test.tsx [hunk 1]
```diff
@@ -23,6 +23,11 @@ describe("pages/lesson-planning.tsx", () => {
 
     expect(screen.getByRole("heading", { level: 1 }).textContent).toBe("hero");
   });
+  it("Renders a nav", () => {
+    render(<PlanALesson pageData={testPlanningPageData} />);
+
+    expect(screen.getAllByText("Contents")).toHaveLength(2);
```
Comment (njblore) [inline src/__tests__/pages/lesson-planning-new.test.tsx:29]: feels like getByRole 'nav' could work here?

**Review by njblore — APPROVED** [review]

**Review by Monika-Koziol — APPROVED** [review]

File: src/__tests__/pages/lesson-planning-new.test.tsx (hunk 1 above)
Comment (njblore) [reply src/__tests__/pages/lesson-planning-new.test.tsx:29]: hmm just looked and it's actually role 'navigation', is that what you tried?

**Review by jack-skerrett-oak — APPROVED** [review]
Some a11y bits!

- Because the nav has been given `tabindex=0`, when tabbing through the page (with and without a screen reader) you are navigated to it even though you cannot interact with it and we don't want to be able to focus non-interactive elements
- I think it would be good to add [`aria-current=page`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current) to the selected link in the nav to indicate to a screen reader user that this element is where you currently are. We are going to be doing a similar thing for the l[esson overview nav](https://www.notion.so/oaknationalacademy/I-want-the-lesson-overview-page-nav-to-be-accessible-6bbde63b08fb499e852b48454dbb4067?pvs=4)
- There are `aria-label`s on the lis, I am not sure what they are doing. This means that a screen reader is currently reading "Lesson planning is important, list item 1". If we wanted to make it really clear these are items in a nav list, we could say `aria-label="Scroll to lesson planning is important"` but I don't think it is needed per se

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
I approved but meant to request changes, whoops!

**Review by jack-skerrett-oak — APPROVED** [review]

**Pull request merged**